### ISSUE TITLE
Area3D with ConcavePolygonShape3D Emits Incorrect Signals

### ISSUE THREAD

**Issue by ATryder** [godotengine/godot#80633]
### Godot version

Godot 4.1.1 stable

### System information

Manjaro Linux - GL Compatibility

### Issue description

When an Area3D is accompanied by a ConcavePolygonShape3D it will emit body_entered and body_exited signals back to back. Given a ConcavePolygonShape3D the Area3D will emit the body_entered signal when the body first touches a face in the ConcavePolygonShape3D. When the body has fully entered the ConcavePolygonShape3D the Area3D will emit the body_exited signal, though the body is actually fully within the bounds of the ConcavePolygonShape3D. Upon exiting the polygon shape the Area3D will again emit the body_entered signal, when the body touches another face on the ConcavePolygonShape3D as it exits the area, then again emit the body_exited signal.

In short, when passing straight through an Area3D using a ConcavePolygonShape3D the Area3D will emit four signals; body_entered, body_exited, body_entered, body_exited, as opposed to just two signals; body_entered, body_exited.

### Steps to reproduce

-Import any mesh from you favorite Modelling suite

-Create a new 3D scene in Godot

-Import the mesh into the scene

-Select the Mesh and then select Mesh --> Create Trimesh Collision Sibling

-Create an Area3D node and make the newly created CollisionShape3D a child of it

-Create a script on any of the scene nodes and connect the Area3D's body_entered and body_exited signals to it

### Minimal reproduction project

[AreaCollision.zip](https://github.com/godotengine/godot/files/12337673/AreaCollision.zip)

**Comment by trvkv** [comment]
Got the same issue. I'm using simple cubes imported from Blender in my project. Collisions are generated via "Create Trimesh Static Body" option available with MeshInstance3D. Area3D are sending unexpected entering/exiting signals, even when Area3D and ColisionShape3D doesn't overlap at all.

I was also trying other collision shapes, but the issues are still there...

Is there some workaround for this? I'm trying to do simple interaction system, meaning, checking with Area3D what's in front of the player.

**Comment by ATryder** [comment]
The work around is to use a convex collision shape or a primitive collision shape.

For complex collision meshes just do your best to approximate the shape with multiple convex meshes then convert each to convex shapes via that same menu you're using or append "-convcol" or "-convcolonly" to the end of your object names in Blender then collect the shapes under the same Area3D in Godot after importing.

If you're just using cubes, a primitive shape should offer better performance. Create an Area3D then a child CollisionShape3D. In the CollisionShape options create a new Box shape and size it to your liking.

**Comment by nargleflex** [comment]
Running into this same issue because I used a CSGPolygon3D and a Path3D to generate & bake a collision mesh, the plan was to create an area using that mesh for player interaction. For what it's worth, this issue is still present in v4.5.dev5.official [64b09905c]

**Comment by arnaucmd** [comment]
is there an easier workaround to this issue? just trying to make complex water volumes and I need to use the same meshes as areas to know if the player is inside... my only solution is to manually split my mesh into multiple convex meshes in blender?
